fastlane/lib/fastlane/actions/slather.rb

```diff
@@ -75,6 +75,17 @@ def self.validate_params!(params)
         else
           UI.user_error!("You have to provide a project with `:proj` or use a .slather.yml")
         end
+
+        # for backwards compatibility when :binary_file type was Boolean
+        if params[:binary_file] == true || params[:binary_file] == false
+          params[:binary_file] = nil
+        end
+
+        # :binary_file validation was skipped for backwards compatibility with Boolean. If a
+        # Boolean was passed in, it has now been removed. Revalidate :binary_file
+        binary_file_options = available_options.find { |a| a.key == :binary_file }
+        binary_file_options.skip_type_validation = false
+        binary_file_options.verify!(params[:binary_file])
```

Comment:
Manually changing `skip_type_validation` and calling `verify!` here kind of spooks me a bit but I guess it makes sense in this case 🙃

Reply:
I had the same feeling, but after thinking a bit this seems the best way to actually be able to update the option to the new type, handle possible old usages, and still make sure the data is properly verified, true? Or did we overcomplicate things?

(A few more comments might be useful that link back to this PR, if future someone wonder wtf is going on here)

Reply:
I'm not a fan of skipping the validation either.  But I don't see anywhere where slather.rb can do parameter house cleaning before the validation takes place without modifying runner.rb.  I'm not familiar enough with the Fastlane code base (or Ruby) to provide any recommendations.  But putting slather specific code in runner.rb felt wrong.

Reply:
@joshdholtz : Just sent you an invite to my fork.  If it's something you're able to add quickly, awesome and much appreciated!  Otherwise, I can take a look at it later tonight.  I definitely agree there should be some tests around this.

Reply:
Ah, thanks! I think there is actually just a check box you could have selected on the right side of this pull request that allows all/any maintainers to edit too 🙃 https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Reply:
I see it's checked now.  Not sure if you had worked your magic or if it was already checked.  Let me know if I need to click/check on anything else...

Reply:
Unit test for true and false added.